src/vs/workbench/contrib/notebook/browser/view/cellParts/cellComments.ts

```diff
@@ -102,25 +101,41 @@ export class CellComments extends CellPart {
 					this._createCommentTheadWidget(info.owner, info.thread);
 					const layoutInfo = (this.currentElement as CodeCellViewModel).layoutInfo;
 					this.container.style.top = `${layoutInfo.outputContainerOffset + layoutInfo.outputTotalHeight}px`;
-
-					this.currentElement.commentHeight = dom.getClientArea(this._commentThreadWidget!.container).height;
-
+					this.currentElement.commentHeight = this._calcualteCommentThreadHeight(this._commentThreadWidget!.getDimensions().height);
 					return;
 				}
 
 				if (this._commentThreadWidget) {
-					if (info) {
-						this._commentThreadWidget.updateCommentThread(info.thread);
-						this.currentElement.commentHeight = dom.getClientArea(this._commentThreadWidget.container).height;
-					} else {
+					if (!info) {
 						this._commentThreadWidget.dispose();
 						this.currentElement.commentHeight = 0;
+						return;
+					}
+					if (this._commentThreadWidget.commentThread === info.thread) {
+						this.currentElement.commentHeight = this._calcualteCommentThreadHeight(this._commentThreadWidget.getDimensions().height);
+						return;
 					}
+
+					this._commentThreadWidget.updateCommentThread(info.thread);
+					this.currentElement.commentHeight = this._calcualteCommentThreadHeight(this._commentThreadWidget.getDimensions().height);
 				}
 			}
 		}));
 	}
 
+	private _calcualteCommentThreadHeight(bodyHeight: number) {
```

Comment:
```suggestion
	private _calculateCommentThreadHeight(bodyHeight: number) {
```

Though other references will also need to be updated.